### PR TITLE
option to disable git packaging for adhoc execution

### LINKED ISSUE #122: Submodules in ad-hoc runs
`git ls-files` does not delve into submodule directories, so when using Git to package files for ad-hoc runs, those files don't get uploaded.

`git ls-files --recurse-submodules` is only supported with `--cached` at present.

A workaround is to hack the packager not to use git for packaging.

Options:

* Expose the `allow_git` flag (e.g. an envvar/global click flag `--no-git`)
* If we'd otherwise use `git ls-files`, detect if submodules are in use and don't use `git ls-files` if they are. (e.g. if the output of `git submodule` is longer than zero).

### PR DESCRIPTION
**Pull request by Kenneth-Schroeder**
should resolve #122

### CODE REVIEW

File: valohai_cli/commands/pipeline/run/run.py [hunk 1]
```diff
@@ -51,7 +53,7 @@ def run(
     if yaml and not adhoc:
         raise click.UsageError('--yaml is only valid with --adhoc')
 
-    commit = create_or_resolve_commit(project, commit=commit, adhoc=adhoc, yaml_path=yaml)
+    commit = create_or_resolve_commit(project, commit=commit, adhoc=adhoc, yaml_path=yaml, allow_git_packaging=not disable_git_packaging)
```
Comment (akx) [inline valohai_cli/commands/pipeline/run/run.py:56]:
```suggestion
    commit = create_or_resolve_commit(project, commit=commit, adhoc=adhoc, yaml_path=yaml, allow_git_packaging=git_packaging)
```

**Review by akx — CHANGES_REQUESTED** [review]
Thanks!

I think a positive argument that's set to `true` by default, and can be disabled with a negative flag, would be more intuitive.

NB: I didn't test the suggestions locally, so they might be wrong :)

**Comment by Kenneth-Schroeder** [comment]
Thank you, good feedback, I am not familiar with click :)

**Review by teroyks — APPROVED** [review]
LGTM, and the test agrees.

**Pull request merged**